src/test/types.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const FILE_EXTENSION = ".linttest";
```

Comment:
Agreed on not ending in `.ts.` The other point of importance is that the first file part of the file extension is significant (`.ts.linttest` or `.tsx.linttest`). As far as the second part, I'm happy with whatever, `.lint` is fine (other than not conveying the intent that it's a test).